src/jarabe/frame/activitiestray.py

```diff
@@ -90,6 +100,53 @@ def __init__(self, home_activity, group):
         elif home_activity.props.launch_status == shell.Activity.LAUNCH_FAILED:
             self._on_failed_launch()
 
+    def drag_drop_cb(self, widget, context, x, y, time):
+        self._drag_save = False
+        context_targets = context.list_targets()
+        for target in context_targets:
+            if str(target) not in ('TIMESTAMP', 'TARGETS', 'MULTIPLE'):
+                widget.drag_get_data(context, target, time)
+        return True
+
+    def journal_drag_data_received_cb(self, widget, context, x, y, selection,
+                              targetType, time):
+        if (hasattr(self,'_drag_save') and self._drag_save == True) or not hasattr(self,'_drag_save'):
+            return
+        logging.debug('Saved to Journal')
+        self._drag_save = True
+        data = None
+        file_path = None
+        if selection.get_text() != None:
+	    data = selection.get_text()
+        if len(selection.get_uris()) != 0:
+            file_path = selection.get_uris()[0]
+        if selection.get_pixbuf() != None:
```

Comment:
Hum, I think that you should change the order of priorities, as the clipboard can have multipul formats in it.

Maybe put images 1st, links 2nd and text 3rd?  Or just do whatever the clipboard does.